File: benchmarks/tasks/fib/fib.c
/* fib n, the naive way: two calls and an addition per node, nothing else. */

#include <stdio.h>
#include <stdint.h>

static int64_t fib(int64_t n) {
    return n < 2 ? n : fib(n - 1) + fib(n - 2);
}

int main(void) {
    printf("%lld\n", (long long)fib(32));
    return 0;
}

File: benchmarks/tasks/fib/Fib.java
// fib n, the naive way: two calls and an addition per node, nothing else.

public class Fib {
    static long fib(long n) {
        return n < 2 ? n : fib(n - 1) + fib(n - 2);
    }

    public static void main(String[] args) {
        System.out.println(fib(32));
    }
}
